sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -483,41 +532,115 @@ public void validate(PipelineOptions options) {
 
     @Override
     public PCollection<TableRow> expand(PBegin input) {
-      final String stepUuid = BigQueryHelpers.randomUUIDString();
-      BoundedSource<TableRow> source;
+      Pipeline p = input.getPipeline();
+      final String staticJobUuid = BigQueryHelpers.randomUUIDString();
+
+      final PCollectionView<String> jobIdTokenView;
+      PCollection<String> jobIdTokenCollection = null;
+      if (!getUseNewSource()) {
```

Comment:
Can you merge this if and the next one at line 561, and then move staticJobUuid into the "if" branch?